### ISSUE TITLE
Error when loading "xiaomi_fan" (after manually installing custom component)

### ISSUE THREAD

**Issue by Zwo12** [syssi/xiaomi_fan#143]
Hi everybody,

when trying to run this integration in Home Assistant (core-2021.4.6) I encouter the following error:

- _"No 'version' key in the manifest file for custom integration 'xiaomi_miio_fan'. As of Home Assistant 2021.6, this integration will no longer be loaded. Please report this to the maintainer of 'xiaomi_miio_fan' "_

followed by: 

- _"Platform error fan.xiaomi_miio_fan - cannot import name 'Fan1C' from 'miio' (/usr/local/lib/python3.8/site-packages/miio/__init__.py) "_

Weird thing about it: 
I neither run the version mentioned in the log nor do I try to import a "Fan1C". I have a zhimi.fan.za5 that I am trying to integrate into docker version of core-2021.4.6

Since I am a total newbie - does anybody have a hint on how to make progress? 
I could not find similar cases searching the web...

Thanks!

**Comment by syssi** [comment]
Please update this custom component (and all other installed custom components interfacing xiaomi devices). This should solve your issue. It's important every component requires the same (and most recent) `python-miio` version. It looks like the python-miio version gets downgraded.

**Comment by Zwo12** [comment]
I downloaded the latest version (0.3.12) and copied it into a clean config (i.e. no other Xiaomi integration nor other component present). 
Then I edited the configuration.yaml according to the setup described in this repo, incl. the correct (I doublechecked!) API token. 
When checking the correctness of the configuration.yaml HA does not finish. Ignoring this check and restarting the server anyways results in the attached error message.
When disabling the corresponding configuration entry the re-start works. So the issue somehow seems to be linked to this component.
I did not specifically check on the python-miio version since I hope that it somehow comes with this release and no other component is installed in parallel. Am I right?
[Error.txt](https://github.com/syssi/xiaomi_fan/files/6682900/Error.txt)

**Comment by syssi** [comment]
Your error.txt contains a syntax error. There is something wrong with your copy of this component. Could you install the custom component from scratch? Using HACS should be the easiest way.

**Comment by Zwo12** [comment]
Clean up of the config folder of the HA instance by removing three rather large files without extension (don't know where they originated from - but they have been created "lately") and the "custom_component" folder resolved the issuer after re-installing the latest version (0.3.13). I guess, messing around with this and another integration while trying to get the fan control working messed up everything (and let to the creation of those mysterious files). I am also using a newer version of HA now (core-2021.6.6), but I think is not part of the solution though. 

As you can tell, I am completely clueless on how I got the issue solved, but at least most of it is working now - awesome!

In enclosed screenshot quite a few internal sensors are not integrated though. Can you confirm that this is the scope it is supposed to be?
![grafik](https://user-images.githubusercontent.com/86188476/124386213-aa69b380-dcd9-11eb-9159-058800543f4d.png)

**Comment by semyag** [comment]
@Zwo12 Can you go past Level 3? mine seems to cap out there and I cannot control the fan any higher than 74% from HA. Same thing with the Oscillation angle. That wont change. The one thing I noticed you and I have the same model which is not  listed as compatible with this integration. My guess is you got the V3?

**Comment by Zwo12** [comment]
@semyag Sorry for the late reply! 
Correct, as you can see in the screenhot my previous post: The speed control (translates "Geschwindigkeit") of the panel is at 100% - the pre-set ("Voreinstellung) indicates level 3 / same with raw speed. So I cannot got past speed level 3 either. 
Nor can I change the oscillation angle. This is because, as I have thought so far, I am just too stupid to program it. 
I have the "Smartmi Standing Fan V3", which is identified as "zhimi.fan.za5, being listed in the supported devices.

**Comment by Zwo12** [comment]
After updating to the current release (0.5.2) I got lots of more information and I can increase the speed up to the max (level 4). Big thanks to all who contributed to this custom component and its ongoing development!
